debug/org.eclipse.debug.ui/ui/org/eclipse/debug/internal/ui/actions/breakpoints/EnableAllBreakpointsAction.java

```diff
@@ -0,0 +1,119 @@
+package org.eclipse.debug.internal.ui.actions.breakpoints;
+
+import org.eclipse.core.resources.IMarkerDelta;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.core.runtime.jobs.Job;
+import org.eclipse.debug.core.DebugPlugin;
+import org.eclipse.debug.core.IBreakpointManager;
+import org.eclipse.debug.core.IBreakpointsListener;
+import org.eclipse.debug.core.model.IBreakpoint;
+import org.eclipse.debug.internal.ui.DebugUIPlugin;
+import org.eclipse.debug.internal.ui.actions.AbstractEnableAllActionDelegate;
+import org.eclipse.debug.internal.ui.actions.ActionMessages;
+import org.eclipse.debug.internal.ui.preferences.IDebugPreferenceConstants;
+import org.eclipse.jface.action.IAction;
+import org.eclipse.jface.dialogs.IDialogConstants;
+import org.eclipse.jface.dialogs.MessageDialogWithToggle;
+import org.eclipse.jface.preference.IPreferenceStore;
+import org.eclipse.ui.IWorkbenchWindow;
+
+public class EnableAllBreakpointsAction extends AbstractEnableAllActionDelegate implements IBreakpointsListener {
+
+	@Override
+	protected boolean isEnabled() {
+		boolean allEnabled = true;
+		final IBreakpointManager breakpointManager = DebugPlugin.getDefault().getBreakpointManager();
+		final IBreakpoint[] breakpoints = breakpointManager.getBreakpoints();
+		if(breakpoints.length > 0) {
+			for (IBreakpoint bp : breakpoints) {
+				try {
+					if(!bp.isEnabled()) {
+						allEnabled = false;
+						break;
+					}
+				} catch (CoreException e) {
+					e.printStackTrace();
```

Comment:
please log exceptions using a logger instead of printStackTrace

Reply:
Changes done